### DISCUSSION TITLE
batch download using single file

### DISCUSSION THREAD

**Discussion by radhgk** [gildas-lormeau/SingleFile#1403] [General]
I am very little knowlege of programming. Currently I have 500 html links which I need to save using singlefile .
Can some one suggest me any script for chrome which can save the HTML one by one with same 5-10 seconds gap so that it does not cause any system to crash .

**Comment by gildas-lormeau** [comment]
Did you try to use the "Batch save" feature in the extension? See the entry `SingleFile > Batch save URLs...` in the context menu.

**Reply by radhgk** [reply]
yes , the website is very very slow and the HTML pages take a very long time like  30 sec to 1 min to load . Its a software help webiste which needs login to see the HTML help pages.
I even tried one more chrome addin " Open Multiple URLs" and then tried to save all tabs using singlefile.
As the webpages take one min to load if I try to open 50 tabs only 40 are loaded properly . 
Same is the case with Bath save option , only 5-6 HTML pages are loaded properly and saved , I think all we need is to add some thing like 30 sec gap or 1 min gap  then atleast most of them will be loaded . Please suggest if any other similar option

**Reply by gildas-lormeau** [reply]
Edit the options file as described here https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options and set `maxParallelWorkers` to `1`. You can also add some delay (in ms) before processing the page with the `delayBeforeProcessing` option.

**Reply by radhgk** [reply]
Thanks a lot , i added one more addin " bulk URL opener "  it has time delay option , then use the save all tab option of "single file "
now I have saved some 3000 html files of the help website so far ,
